source/exe/main_common.cc

```diff
@@ -83,9 +83,13 @@ int main_common(OptionsImpl& options) {
   DefaultTestHooks default_test_hooks;
   ThreadLocal::InstanceImpl tls;
   Stats::ThreadLocalStoreImpl stats_store(stats_allocator);
-  Server::InstanceImpl server(options, local_address, default_test_hooks, *restarter, stats_store,
-                              access_log_lock, component_factory, tls);
-  server.run();
+  try {
+    Server::InstanceImpl server(options, local_address, default_test_hooks, *restarter, stats_store,
+                                access_log_lock, component_factory, tls);
+    server.run();
+  } catch (const EnvoyException& e) {
+    return 1;
+  }
   ares_library_cleanup();
```

Comment:
it doesn't really matter that much, but we should probably do this in all cases. I don't think it's worth fixing this though unless you want to.

Reply:
Oh, true enough. Might as well fix it, stand by.